great_expectations/datasource/util.py

```diff
@@ -2,16 +2,45 @@
 
 logger = logging.getLogger(__name__)
 
-try:
-    from pyspark import SparkContext
-    from pyspark.sql import SparkSession
 
-    from great_expectations.datasource.sparkdf_datasource import SparkDFDatasource
-except ImportError:
-    SparkContext = None
-    SparkSession = None
-    SparkDFDatasource = None
-    # TODO: review logging more detail here
-    logger.debug(
-        "Unable to load pyspark; install optional spark dependency for support."
+# noinspection PyPep8Naming
+def get_or_create_spark_session(
+    name: str = "default_great_expectations_spark_dataframe_datasource",
+):
+    # Due to the uniqueness of SparkContext per JVM, it is impossible to change SparkSession configuration dynamically.
+    # Attempts to circumvent this constraint cause "ValueError: Cannot run multiple SparkContexts at once" to be thrown.
+    # Hence, SparkSession with SparkConf acceptable for all tests must be established at "pytest" collection time.
+    # This is preferred to calling "return SparkSession.builder.getOrCreate()", which will result in the setting
+    # ("spark.app.name", "pyspark-shell") remaining in SparkConf statically for the entire duration of the "pytest" run.
```

Comment:
Why are we modifying the primary module to support pytest? What is source doing?

What happens if a user has two datasources configured, with two different sparkconf objects?

Reply:
@jcampbell Thanks for your comments.  The primary reason for the changes proposed in this PR is captured in the above comment: For `PySpark-3`, due to the uniqueness of SparkContext per JVM, it is impossible to change SparkSession configuration dynamically.  It is for this reason that `PR 2090` fails tests in `tests/integration/spark/test_spark_config.py`.  Once we agree on the course of action given the above, I can follow up on your other comments.  Happy to discuss.  Thank you.